react-app/src/components/SettingsModal/index.js

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ActionIcon from 'components/ActionIcon';
+import BaseModal from 'components/BaseModal';
+import Logo from 'components/Logo';
+import RadioButton from 'components/RadioButton';
+import SettingsOptionLabel from 'components/SettingsOptionLabel';
+
+import styles from './index.module.scss';
+
+export default class SettingsModal extends React.PureComponent {
+  static propTypes = {
+    /**
+     * The username to display
+     */
+    username: PropTypes.string.isRequired,
+
+    /**
+     * The function to call when hide icon is tapped
+     */
+    onHide: PropTypes.func.isRequired,
```

Comment:
We should explain here that this function passes the notification settings as argument.

Reply:
Ok.